loleaflet/admin/src/AdminSocketLog.js

```diff
@@ -100,11 +103,64 @@ var AdminSocketLog = AdminSocketBase.extend({
 		document.getElementById('update-log-levels').classList.add('is-info');
 	},
 
+	applyChannelFilter: function() {
+		if (document.getElementById('log-channel-filter').selectedIndex !== 0) {
+			var filteredChannel = document.getElementById('log-channel-filter').value;
+			var filteredLines = '';
+			var lineList = this._logLines.split('\n');
+			for (var i = 0; i < lineList.length; i++) {
+				if (lineList[i].split('[').length > 1) {
+					if (lineList[i].split('[')[1].split(']')[0].trim() === filteredChannel)
+						filteredLines += lineList[i] + '\n';
+				}
+			}
+			document.getElementById('log-lines').value = filteredLines;
+		}
+		else {
+			document.getElementById('log-lines').value = this._logLines;
+		}
+	},
+
+	refreshChannelFilter: function() {
+		var lineList = this._logLines.split('\n');
+		var channelList = new Array(0);
+		for (var i = 0; i < lineList.length; i++) {
+			if (lineList[i].trim() !== '' && lineList[i].split('[').length > 1) {
+				var channelName = lineList[i].split('[')[1].split(']')[0].trim();
+				if (!channelList.includes(channelName))
+					channelList.push(channelName);
+			}
+		}
+
+		var currentFilteredChannel = document.getElementById('log-channel-filter').value;
+
+		// Remove previous channels.
+		for (i = document.getElementById('log-channel-filter').options.length - 1; i > 0; i--)
+			document.getElementById('log-channel-filter').remove(i);
+
+		// Now add new ones.
+		for (i = 0; i < channelList.length; i++) {
+			var option = document.createElement('option');
+			option.text = channelList[i];
+			document.getElementById('log-channel-filter').add(option);
+		}
+
+		// Does previously selected channel name still exist? If not, we will re-set filter.
+		if (!channelList.includes(currentFilteredChannel))
```

Comment:
I think all JS code (in this project) use braces for all code blocks. Best to stay consistent, if that's the case. Same for "else" below.

Reply:
It is also preferred in Javascript.

Reply:
Yes, so why not add braces here?